node/mock_test.go

```diff
@@ -18,11 +17,48 @@ var (
 	_ PodNotifier         = (*mockProvider)(nil)
 )
 
+type waitableInt struct {
+	cond *sync.Cond
+	val  int
+}
+
+func newWaitableInt() *waitableInt {
+	return &waitableInt{
+		cond: sync.NewCond(&sync.Mutex{}),
+	}
+}
+
+func (w *waitableInt) read() int {
+	defer w.cond.L.Unlock()
+	w.cond.L.Lock()
+	return w.val
+}
+
+func (w *waitableInt) until(f func(int) bool) {
```

Comment:
Can this leak goroutines in failure cases?
Seems like we may need a way to cancel this.